capstone-frontend/src/components/AdminClubPage.jsx

```diff
@@ -65,13 +65,12 @@ class AdminClubPage extends Component {
     }
     const history = this.props.history;
     fetch(`/api/clubs?id=${this.props.match.params.id}`, { method: 'delete' })
-      .then(function () {
-        history.push('/myclubs');
-      })
+      .then(history.push('/myclubs'))
```

Comment:
This will cause the navigation to happen immediately after the DELETE request goes out (before you get a response) since it's not inside a function anymore. It might be OK to do this kind of "optimistic" update on the front-end, but you might also need to add some logic to make sure the MyClubs page always has the most up-to-date data (i.e. once the delete has completed, since you could be dealing with a race condition). Otherwise, you can wrap the `history.push` in an arrow function here.

Reply:
I didn't realize that, thanks for pointing it out. I'll revert this to rely on successful completion before pushing like you mentioned.